app/resources/admin.html.template

```diff
@@ -138,37 +203,94 @@
+  function add_language() {
+    add_language_to_repo_titles_table();
+    for(var id_num = 0; id_num < MULTILINGUAL_TABLE_IDS.length; id_num++){
+      add_language_to_custom_message_table(
+          MULTILINGUAL_TABLE_IDS[id_num], 'not selected');
+    }
+  }
+
   /* Hides a language from the user's view and marks it so that it will not be
    * submitted.
    */
   function remove_language(row_number) {
-    var table = $('language_table');
-    var row = table.rows[row_number];
-    row.hidden = true;
-    $('language_selection_' + row_number).value = 'unused';
+    $('repo_titles_language_selection_' + row_number).value = 'unused';
+    var repo_table = $('repo_titles_table');
+    repo_table.rows[row_number].hidden = true;
+    for (var id_num = 0; id_num < MULTILINGUAL_TABLE_IDS.length; id_num++) {
+      var table_id = MULTILINGUAL_TABLE_IDS[id_num];
+      var table = $(table_id + '_table');
+      var row = table.rows[row_number];
+      row.hidden = true;
+    }
   }
 
   // Adds rows to the languages table for each language already in the database
   function add_initial_languages() {
     // Saved languages and titles for this repository
+    var all_view_config = {{all_view_config_json|safe|default:'[]'}};
     var LANGUAGES = {{view_config_json.language_menu_options|safe|default:'[]'}};
-    var TITLES = {{view_config_json.repo_titles|safe|default:'[]'}};
-
-    //foreach language, add that language and its title
+    //foreach language, add that language, its title and optional custom messages
+    var repo_title_values = all_view_config['repo_titles']
     for (var lang_num = 0; lang_num < LANGUAGES.length; lang_num++) {
-      var row_number = add_language();
+      var row_number = add_language_to_repo_titles_table();
       var language = LANGUAGES[lang_num];
-      $('language_selection_' + row_number).value = language;
-      if (TITLES[language] != undefined) {
-        $('language_title_' + row_number).value = TITLES[language];
+      $('repo_titles_language_selection_' + row_number).value = language;
+      if (repo_title_values[language] !== undefined) {
+        $('repo_titles_language_value_' + row_number).value =
+            repo_title_values[language];
+      }
+    }
+
+    for (var id_num = 0; id_num < MULTILINGUAL_TABLE_IDS.length; id_num++) {
+      var table_id = MULTILINGUAL_TABLE_IDS[id_num];
+      var table_values = all_view_config[table_id];
+      for (var lang_num = 0; lang_num < LANGUAGES.length; lang_num++) {
+        var language = LANGUAGES[lang_num];
+          var row_number = 
+              add_language_to_custom_message_table(table_id, language);
+        if (table_values !== undefined && table_values[language] !== undefined) {
+          $(table_id + '_language_value_' + row_number).value
+              = table_values[language];
+        }
       }
     }
+
+    var language_num = $('repo_titles_table').rows.length - 1;
+    for (var row_num = 0; row_num < language_num; row_num++) {
```

Comment:
It would be easier to read if this row_num iterates from 1 to $('repo_titles_table').rows.length - 1 instead. Then you can remove these -1, +1.